core/amber/src/main/scala/edu/uci/ics/amber/engine/architecture/common/WorkflowActor.scala

```diff
@@ -0,0 +1,19 @@
+package edu.uci.ics.amber.engine.architecture.common
+
+import akka.actor.{Actor, ActorLogging, Stash}
+import com.softwaremill.macwire.wire
+import edu.uci.ics.amber.engine.architecture.messaginglayer.MessagingManager
+import edu.uci.ics.amber.engine.architecture.receivesemantics.FIFOAccessPort
+import edu.uci.ics.amber.engine.common.ambermessage.WorkerMessage.DataMessage
+
+class WorkflowActor extends Actor with ActorLogging with Stash {
+
+  lazy val input: FIFOAccessPort = wire[FIFOAccessPort]
+  lazy val messagingManager: MessagingManager = wire[MessagingManager]
+
+  override def receive: Receive = {
+    case msg: DataMessage =>
+      messagingManager.receiveMessage(msg, sender)
+    case other =>
+  }
```

Comment:
Not being used right now. Will use it after StateTransition is implemented in Worker and Controller

Reply:
What do you mean by "not being used"? I see the worker actors extend this class. And how is it related to StateTransitions?

Reply:
Workers are extending the class. But the receive function is not being used right now. They have their own receive functions depending on the state they are in. When we have a single function for all states, it will be easier to abstract the messaging layer functionality to this receive, and only push actual message handling to workers